### ISSUE TITLE
There is no attribute "id" in any tag definition

### ISSUE THREAD

**Issue by up-to-you** [Kotlin/kotlinx.html#66]
Actual workaround : 

```kotlin
div(.....) {
   attributes["id"] = "special-div"
}
```

Expected api : 

```kotlin
div(id = "special-div".....) {}
```

**Comment by cy6erGn0m** [comment]
```
div {
    id = "special-div"
}
```

**Comment by cy6erGn0m** [comment]
In fact `id` is not so frequently used to provide it in _every_ function as an optional parameter however the corresponding attribute is always available so there is no need to use `attributes` map

**Comment by Tatskaari** [comment]
Hi, just had a similar confusion and came up with this helper function for divs: 
```

import org.testng.annotations.Test
import kotlinx.html.*
import kotlinx.html.stream.appendHTML
import java.io.StringWriter
import kotlin.test.assertEquals


object HTMLTest {
  fun FlowContent.div(classes: String = "", vararg attributes: Pair<String, String>, block: FlowContent.() -> Unit) {
    val attributeMap = attributes.toList()
      .plus("class" to classes)
      .associate { it }
    DIV(attributeMap, consumer).visit(block)
  }
  
  

  @Test
  fun `test html with attribute list`(){
    val stringWriter = StringWriter()
    stringWriter.appendHTML().html {
      body {
        div ("main", "id" to "main-body", "style" to "float: left") {
          text("main")
        }
      }
    }
    assertEquals("""
<html>
  <body>
    <div id="main-body" style="float: left" class="main">main</div>
  </body>
</html>""".trim(), stringWriter.toString().trim())
  }
}
```

What do you think? I find the pattern a bit more intuitive (especially for people less familiar with Kotlin) than accessing fields of the tag object via extension method magic.

**Comment by 44x1carbon** [comment]
@Tatskaari 
I think that your proposal is very good as it approaches the original way of writing HTML.
I think that it is better to include ```classes``` in ```attributes``` as well as how to write HTML more.

**Comment by fatjoem** [comment]
I am currently converting the templates for a large website to kotlinx.html. Those templates are using id and also class attributes a lot. Being able to add classes as a parameter but not the id, feels very inconsistent and unsexy.

In a new project I would try to minimize using id attributes, hence this will not be a big issue. But for converting existing projects, it is a big pain.

**Comment by ScottPierce** [comment]
@cy6erGn0m Agreed with the above devs. I use `id`s all the time, and having to declare them  as part of the builder instead of in the tag function is inconsistent. Ideally there would be an optional parameter that I could use:

```
section(id = "aboutSection") {
}
```

**Comment by ScottPierce** [comment]
Over the past several weeks, I've created an alternative library, [kotlin-html](https://github.com/ScottPierce/kotlin-html) to remedy pain points of kotlinx.html that I've found like this; primarily to add css, provide first-class support for ids, classes, and styles, and provide a simpler architecture so it's easier for people to contribute.
